### PR TITLE
Use platform specific implementation of GetCurrentProcessId

### PR DESCRIPTION
**Pull request by marek-safar**
directly in shared location instead of relying on runtime export

### CODE REVIEW

**Comment by jkotas** [comment]
`src/System/Diagnostics/Eventing/EventPipeController.cs(215,51): error CS0117: 'Interop.Kernel32' does not contain a definition for 'GetCurrentProcessId'`

**Comment by marek-safar** [comment]
@jkotas the build issue resolved

File: src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.Windows.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+#if ES_BUILD_STANDALONE
+namespace Microsoft.Diagnostics.Tracing
+#else
+namespace System.Diagnostics.Tracing
+#endif
+{
+    public partial class EventSource : IDisposable
+    {
+        internal static uint GetCurrentProcessId() => Interop.Kernel32.GetCurrentProcessId();
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.Windows.cs]: It does not sound right for `EventSource` to be the place one goes to get the current process ID.

I would put the shared stub into `Interop` file, e.g. similar to what we do for RandomBytes:

https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/Interop/Windows/BCrypt/Interop.BCryptGenRandom.GetRandomBytes.cs#L11
https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/Interop/Unix/System.Native/Interop.GetRandomBytes.cs#L17

**Comment by jkotas** [comment]
Could you please also add the line to https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/StubEnvironment.cs#L379 to make CoreFX build fine once this mirrors over?

**Pull request merged**